R/rdeephaven/R/table_handle_wrapper.R

```diff
@@ -1,43 +1,91 @@
-# TODO: Document this guy
-
+#' @title Deephaven TableHandles
+#' @description Deephaven TableHandles are references to tables living on a Deephaven server. They provide an
+#' interface for interacting with tables on the server.
+#' 
+#' @usage NULL
+#' @format NULL
+#' @docType class
+#' 
+#' @section Methods
+#' 
+#' - `$bind_to_variable(name)`
+#' - `$to_arrow_record_batch_stream_reader()`
+#' - `$to_arrow_table()`
+#' - `$to_tibble()`
+#' - `$to_data_frame()`
+#'
+#' @examples
+#' 
+#' # connect to the Deephaven server running on "localhost:10000" using anonymous 'default' authentication
+#' client_options <- ClientOptions$new()
+#' client <- Client$new(target="localhost:10000", client_options=client_options)
+#' 
+#' # open a table that already exists on the server
+#' new_table_handle1 <- client$open_table("table_on_the_server")
+#' 
+#' # convert the Deephaven table to an R data frame
+#' new_data_frame <- new_table_handle1$to_data_frame()
+#' 
+#' # modify new dataframe in R
+#' new_data_frame$New_Int_Col <- c(1, 2, 3, 4, 5)
+#' new_data_frame$New_String_Col <- c("I", "am", "a", "string", "column")
+#' 
+#' # push new dataframe to the server and name it "new_table"
+#' new_table_handle2 <- client$import_table(new_data_frame)
+#' new_table_handle2$bind_to_variable("new_table")
 
 TableHandle <- R6Class("TableHandle",
     public = list(
 
         initialize = function(table_handle) {
-
             if (class(table_handle)[[1]] != "Rcpp_INTERNAL_TableHandle") {
                 stop("'table_handle' should be an internal Deephaven TableHandle. If you're seeing this,
                 you are trying to call the constructor of TableHandle directly, which is not advised.")
             }
             private$internal_table_handle <- table_handle
         },
 
+        #' @description
+        #' Binds the table referenced by this TableHandle to a variable on the server,
+        #' enabling it to be accessed by that name from any Deephaven API.
+        #' @param name New name for this table on the server.
```

Comment:
```suggestion
        #' @param name Name for this table on the server.
```
What if the table name is overwritten?  Then it is not new.